src/Author.java

```diff
@@ -15,19 +17,18 @@ public String getLastName() {
         return lastName;
     }
 
-    public void setName(String name) {
-        this.name = name;
-    }
-
-    public void setLastName(String lastName) {
-        this.lastName = lastName;
-    }
     @Override
     public String toString() {
         return name + " " + lastName;
     }
     @Override
     public int hashCode(){
-        return java.util.Objects.hashCode(toString());
+        return java.util.Objects.hashCode(this.name) & java.util.Objects.hashCode(this.lastName);
```

Comment:
не нужно использовать побитовые операции, надо просто передавать все поля в методе hashCode
```suggestion
        return java.util.Objects.hashCode(name, lastName);
```